codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/ServerCodegenVisitor.kt

```diff
@@ -444,3 +463,48 @@ open class ServerCodegenVisitor(
             .render()
     }
 }
+
+fun renderDefaultBuilder(model: Model, symbolProvider: RustSymbolProvider, member: MemberShape): Writable {
+    return writable {
+        val node = member.expectTrait<DefaultTrait>().toNode()!!
+        when (val target = model.expectShape(member.target)) {
+            is EnumShape, is IntEnumShape -> {
+                val value = when (target) {
+                    is IntEnumShape -> node.expectNumberNode().value
+                    else -> node.expectStringNode().value
+                }
+                val enumValues = when (target) {
+                    is IntEnumShape -> target.enumValues
+                    is EnumShape -> target.enumValues
+                    else -> mapOf<String, String>()
+                }
+                val variant = enumValues
+                    .entries
+                    .filter { entry -> entry.value == value }
+                    .map { entry ->
+                        symbolProvider.toEnumVariantName(
+                            EnumDefinition.builder().name(entry.key).value(entry.value.toString()).build(),
+                        )!!
+                    }
+                    .first()
+                val symbol = symbolProvider.toSymbol(target)
+                rust(".unwrap_or($symbol::${variant.name})")
+            }
+            is IntegerShape -> rust(".unwrap_or(${node.expectNumberNode().value})")
+            is FloatShape -> rust(".unwrap_or(${node.expectNumberNode().value.toFloat()})")
+            is BooleanShape -> rust(".unwrap_or(${node.expectBooleanNode().value})")
+            is StringShape -> rust(".unwrap_or_else(|| String::from(${node.expectStringNode().value.dq()}))")
+            is ListShape, is MapShape -> rust(".unwrap_or_default()")
```

Comment:
This is incorrect. In the case of list and map shapes, the default trait's value should be a document. We need to parse said document node and construct the equivalent value in Rust.

Reply:
From the [docs](https://smithy.io/2.0/spec/type-refinement-traits.html#default-trait) I read:

> The following shapes have restrictions on their default values:
> ...
> list: can only be set to an empty list.
> map: can only be set to an empty map.

Where is the different default?

Reply:
You're right! My bad, I hadn't read that section. I had assumed that since the value for the `@default` trait was a document, it could be arbitrarily complex.